ui/src/components/layout/FlowCanvas.tsx

```diff
@@ -127,15 +168,17 @@ const FlowCanvas: React.FC<Props> = () => {
   }
 
   function handleAddNode(item: ThingInput) {
-    // Item is whatever has just been entered in the modal
+    // Item is from the form entered in the modal
     // and has already been stored on the DHT by this point
 
     // Create react flow node object
+    const id = Guid.raw();
```

Comment:
This shouldn't be necessary, since a call to new DisplayNode() without a defined id should automatically create one.